src/main/java/burger/list/List.java

```diff
@@ -0,0 +1,223 @@
+     *
+     * @param event the event input by the user.
+     * @throws ArrayIndexOutOfBoundsException if deadline format is not followed.
+     */
+    public void addEvent(String[] event) {
+        try {
+            StringBuilder eventText = new StringBuilder(event[1]);
+            int i = 2;
+            while (!event[i].startsWith("/from")) {
+                eventText.append(' ').append(event[i]);
+                i++;
+            }
+            eventText.append(' ').append("(from: ");
+            i++;
+            while (!event[i].startsWith("/to")) {
+                eventText.append(event[i]).append(' ');
+                i++;
+            }
+            eventText.append("to: ");
+            i++;
+            while (i < event.length - 1) {
+                eventText.append(event[i]).append(' ');
+                i++;
+            }
+            eventText.append(event[i]).append(')');
+            Task newEvent = new Event(eventText.toString());
+            addTask(newEvent);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            printInvalidEventInputMessage();
+        }
+    }
+
+    /**
+     * Adds a new deadline to the list.
+     *
+     * @param deadline the deadline input by the user.
+     * @throws ArrayIndexOutOfBoundsException if deadline format is not followed.
+     */
+    public void addDeadline(String[] deadline) {
+        try {
+            StringBuilder deadlineText = new StringBuilder(deadline[1]);
+            int i = 2;
+            while (!deadline[i].startsWith("/")) {
+                deadlineText.append(' ').append(deadline[i]);
+                i++;
+            }
+            deadlineText.append(' ').append("(by: ");
+            i++;
+            while (i < deadline.length - 1) {
+                deadlineText.append(deadline[i]).append(' ');
+                i++;
+            }
+            deadlineText.append(deadline[i]).append(')');
+            Task newDeadline = new Deadline(deadlineText.toString());
+            addTask(newDeadline);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            printInvalidDeadlineInputMessage();
+        }
+    }
+
+    /**
+     * Adds task to the list and prints out a message.
+     *
+     * @param task Task that is being added.
+     */
+    public void addTask(Task task) {
+        this.taskList.add(task);
+        printLine();
+        System.out.println("Got it. I've added this task:");
+        this.taskList.get(totalTasks).printTask();
+        this.totalTasks++;
+        System.out.println();
+        System.out.println("Now you have " + totalTasks + " tasks in the list.");
+        printLine();
+    }
+    /**
+     * Returns the mark from a task in the list.
+     *
+     * @param idx the index of the task.
+     * @param command Command to "mark" or "unmark" the task.
+     * @throws BurgerException if index input is out of the range of list of tasks.
+     */
+    public void getMark(int idx, String command) {
```

Comment:
the name of this function and the description do not quite reflect what is being done in the function.

you are also marking tasks in the same function, unlike how it has been described